.ci/opensearch/Dockerfile

```diff
@@ -1,9 +1,10 @@
-ARG STACK_VERSION
-FROM "opensearchproject/opensearch:$STACK_VERSION"
+ARG SOURCE_IMAGE
+FROM $SOURCE_IMAGE
 
-RUN /usr/share/opensearch/opensearch-onetime-setup.sh
-COPY esnode.pem /usr/share/opensearch/config/esnode.pem
-COPY esnode-key.pem /usr/share/opensearch/config/esnode-key.pem
+ARG SECURE_INTEGRATION
+
+COPY --chown=opensearch:opensearch *.yml *.pem /usr/share/opensearch/config/
```

Comment:
I think the previous way to tailor the images is much simpler: we don't need to introduce different yml files for secure / non-secure cluster etc,  why not to keep it that way? To me looks like more burden to support ...

Reply:
The reason for splitting the config in two and then appending them, is due to the fact the unreleased OpenSearch doesn't have the security plugin and so errors out when it hits the unknown `plugins.security.*` settings in yaml. So there's no "matrix" of combinations here, so I don't see it as a significant burden when it allows having a single source of truth for settings between local dev and GitHub actions without them being encoded in a shell script. It also means if you need to do any debugging with this image the config is baked in without having to worry about runtime setting them/repeatedly running the script.

There is also an alternative of using a single config file and doing a `grep -v 'plugins.security'` to get a clean config for unreleased etc.

Reply:
> The reason for splitting the config in two and then appending them, is due to the fact the unreleased OpenSearch doesn't have the security plugin and so errors out when it hits the unknown plugins.security.* settings in yaml.

I think that's kind of fine, for the client it just means to use HTTP (not HTTPs). In these regards, the client is agnostic to whatever configuration is there (w/o security, but it comes with image), it just needs to know the URL.

Reply:
> > The reason for splitting the config in two and then appending them, is due to the fact the unreleased OpenSearch doesn't have the security plugin and so errors out when it hits the unknown plugins.security.* settings in yaml.
> 
> I think that's kind of fine, for the client it just means to use HTTP (not HTTPs). In these regards, the client is agnostic to whatever configuration is there (w/o security, it comes with image), it just needs to know the URL.

No the "unreleased" OpenSearch build is a minimal distribution, so is missing the security plugin, as such the cluster/node will fail to start if you specify the `plugins.security.*` settings

Reply:
That is correct because the `plugins.security.*` settings are set manually now [1], that basically circles back to initial question - do we need that? 

The way it works now (on `main`) is very simple:
 - for released versions - update certs, keep default configs, enable / disable security, run w/o HTTPS
 - for unreleased, keep default configs, run HTTP only

My apologies if I am missing something here, but that looks simple enough, the configuration part becomes more complicated with this change (just a subjective opinion for sure).

[1] https://github.com/opensearch-project/opensearch-rs/pull/128/files#diff-24fc03647d5062ea4a83520481cdc2df7039f5d3061a0c7c62ab8eabe405e2e0

Reply:
> And either way we don't currently only use default settings either, they're just encoded in a shell script and set at runtime on the docker container https://github.com/opensearch-project/opensearch-rs/blob/main/.ci/run-opensearch.sh#L23-L32

That's perfectly fine (and very convenient to be fair), I would prefer to keep it this way instead of modifying container's `opensearch.yml`. The huge benefit here - you don't need to dial with `plugin.security.*` settings anymore.

Reply:
>  and as there's still a plugins.security.* setting that needs overriding.

You mean `plugin.security.audit.type`?

Reply:
> > and as there's still a plugins.security.* setting that needs overriding.
> 
> You mean `plugin.security.audit.type`?

Yes

Reply:
Any particular strong reasons to change it vs keeping  whatever is the default?

Reply:
> Any particular strong reasons to change it vs keeping whatever is the default?

It accounts for 80% of the speedup on the secured tests, as default is `internal_opensearch` but its target index doesn't exist and isn't auto created so just errors continuously.

Reply:
Oh ... I see, thanks a lot for clarifying